Mobile-configuration.md

```diff
@@ -0,0 +1,81 @@
+# Standarize mobile configuration format
+
+## Introduction
+
+Introduce standard for system interoperability, by providing schemas that will be used to communicate between mobile sdk's and core services.
+
+## Problem Description
+
+Mobile configuration format is not standarized. It contains various fields that have no meaning for the mobile users.
+Configuration also exposes some internal fields. 
+
+```json
+{
+ "config": {
+          "headers": {},
+          "mysql_db": "unifiedpush",
+          "mysql_password": "unifiedpush",
+          "mysql_username": "unifiedpush",
+          "name": "unified-push-server",
+          "type": "unified-push-server",
+          "uri": "http://ups-myproject.192.168.37.1.nip.io"
+        },
+        "name": "unified-push-server"
+}
+```
+
+Most important parameters often have different names. For example some services contain `url` property and some `uri`.
+Service specific metadata prevents us from strict json parsing.
+
+## Expectations
+
+- Provide mobile configuration that has only esential fields for mobile users
+- Standarize format so users to not break functionalities once users will start using configuration
+ 
+## Proposed Solution
+
+Provide standard for mobile configuration that will be extendable and will be easy to parse in mobile clients.
+
+### Mobile cli specifics
+
+Mobile-cli currently print configuration out to the std. 
+
+### Contract between SDK and AeroGear Services Backend
+
+AeroGear services cli (and maybe UI) should clearly define format for the interaction with the SDK.
+To do that efficiently we suggest to use JSONSchema format with versioning.
+
+Each backend service will need to conform to the schema in order to get supported.
+Schema can be also used as documentation for services. Backend developers can iterate on schema by changing schema versions and notifying SDK developers to adjust SDK to conform with the new format.
+
+
+### Schema overview
+
+`mobile-service.json` file will aggregate all service definitions.
+Each service will have their own specific data (schema) that will be used directly in service SDK.
+This schema will be supplied in additional `metadata` parameter.
+
+### Initial top level json schema
+
+#### Top level object
+
+```json
+{
+ "version" : "number",
+ "serverUrl" : "string",
```

Comment:
perhaps change to cluster name or identifier. Not sure exposing this to all clients is what end users would want